src/main/java/com/_up/megastore/services/implementations/SizeService.java

```diff
@@ -41,4 +43,19 @@ public SizeResponse updateSize(UUID sizeId,UpdateSizeRequest updateSizeRequest){
 
         return SizeMapper.toSizeResponse(sizeRepository.save(size));
     }
+
+    @Override
+    public SizeResponse restoreSize(UUID sizeId){
+        validateSizeForRestoration(sizeId);
+        Size size = findSizeByIdOrThrowException(sizeId);
```

Comment:
Siguiendo con el comentario anterior acá habría que invertir el orden

```suggestion
        Size size = findSizeByIdOrThrowException(sizeId);
        validateSizeForRestoration(size);
```